app/controllers/orders/client_summary.js

```diff
@@ -17,29 +22,80 @@ export default detail.extend({
 
   identityTypes: Ember.computed(function() {
     return [
-      { name: 'Hong Kong Identity Card', id: 1 },
-      { name: 'Asylum Seeker Recognizance Form', id: 2}
+      { name: "Hong Kong Identity Card", id: 1 },
+      { name: "Asylum Seeker Recognizance Form", id: 2 }
     ];
   }),
 
+  isLongDescription: Ember.computed("model.purposeDescription", function() {
+    return this.get("model.purposeDescription.length") > LONG_DESCRIPTION_SIZE;
+  }),
+
+  toggleDisplayShowMore: Ember.observer("isLongDescription", function() {
+    const isLongDescription = this.get("isLongDescription");
+    isLongDescription
+      ? this.set("displayShowMore", true)
+      : this.set("displayShowMore", false);
+  }),
+
+  shortPurposeDescription: Ember.computed(
+    "showInlineEdit",
+    "displayShowMore",
+    function() {
+      let purposeDescription = this.get("model.purposeDescription");
+      if (this.get("displayShowMore")) {
+        return purposeDescription.substring(0, LONG_DESCRIPTION_SIZE) + "...";
+      } else {
+        return purposeDescription;
+      }
+    }
+  ),
+
   actions: {
+    showMoreToggle() {
+      this.toggleProperty("displayShowMore");
+    },
+
+    enableDescriptionEdit(opts = {}) {
+      const { makeEditable = true } = opts;
+      this.set("showInlineEdit", makeEditable);
+      if (makeEditable) {
+        Ember.run.later(() => {
+          Ember.$("#desc").click();
+        }, 50);
+      }
+    },
+
+    updatePurposeDescription(orderId, body) {
```

Comment:
If this is named `updatePurposeDescription`, then it should probably take a "description" argument instead of "body"

Note that you could also do
```javascript
order.set('purposeDescription', description)
order.save()
```